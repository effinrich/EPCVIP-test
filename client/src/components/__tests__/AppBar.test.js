import React from 'react'
import { Box, Image, Button } from 'grommet'
import { Menu } from 'grommet-icons'

import { render, screen } from 'utils/testUtils/helperRtl'
import Logo from 'assets/logo.png'

import AppBar from '../AppBar'

const children = () => (
  <div>
    <Box>
      <Image
        fill
        alt="logo"
        src={Logo}
        style={{ width: '100%', maxWidth: '170px', height: 'auto' }}
      />
    </Box>

    <Box>
      <Button plain icon={<Menu size="medium" />} />
    </Box>
  </div>
)

describe('AppBar Component', () => {
  it('should render without crashing', () => {
    const { container } = render(<AppBar />)
    expect(container).toMatchSnapshot()
  })

  it('should render a logo image', () => {
    const { getByAltText } = render(<AppBar children={children()} />)
    expect(getByAltText(/logo/i)).toBeInTheDocument()
  })

  it('should render a menu button', () => {
    render(<AppBar children={children()} />)
    expect(screen.getByRole('button', { name: /Menu/i })).toBeInTheDocument()
  })
})
