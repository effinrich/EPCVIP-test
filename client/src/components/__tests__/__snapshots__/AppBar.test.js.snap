// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Component should render without crashing 1`] = `
<div>
  <header
    class="StyledBox-sc-13pk1d4-0 klPIBY"
    style="z-index: 1; min-height: 65px;"
  />
</div>
`;
